### ISSUE TITLE
Block editing of cacerts in UI/API

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#15273]
The contents of `/v3/cacerts` should not be manually edited. It's either set to the value of the self signed generated certificate, or the bind mounted certificate, or set to empty when `--no-cacerts` is passed. Editing the value manually is not supported and cause issues.

**Comment by davidnuzik** [comment]
It is still possible to edit the cacerts via the API in the UI. You can modify the string and send the request and successfully manually modify the cacerts contents manually.

**Comment by cjellick** [comment]
fixed in UI but not backend. I've removed the UI label.

**Comment by loganhz** [comment]
It has been blocked in API as well

**Comment by davidnuzik** [comment]
Version: Master 2.2 (1/28/19)
Confirmed fix. I can now no longer edit the cacerts value in the UI API. Upon doing so I appropriately get a 405 error with a message of "cacerts is readOnly". I also cannot edit the cacerts value in the UI on the Settings page.

**Comment by cjellick** [comment]
Have to revert this fix. Reverting

**Comment by davidnuzik** [comment]
Version: Master 2.2 (2/14/19)
Confirmed fix. You cannot modify the contents of cacerts in the UI. The function menu only has a "View in API" option. When I view in API I can see that there is no update option and I cannot edit the value here either.

Upon attempting to do so I get an HTTP 405 with code "MethodNotAllow" as expected.
```
{
"baseType": "error",
"code": "MethodNotAllow",
"message": "cacerts is readOnly",
"status": 405,
"type": "error"
}
```

**Comment by tillo** [comment]
It seems to me that this is the only barrier to having TLS offloading on a reverse proxy in front of Rancher for agents to reach it.

```
INFO: https://rancher.home.tillo.ch/ping is accessible
INFO: rancher.home.tillo.ch resolves to 178.193.84.99
INFO: Value from https://rancher.home.tillo.ch/v3/settings/cacerts is an x509 certificate
time="2024-09-10T08:10:07Z" level=info msg="Listening on /tmp/log.sock"
time="2024-09-10T08:10:07Z" level=info msg="Rancher agent version v2.9.1 is starting"
time="2024-09-10T08:10:07Z" level=info msg="Testing connection to https://rancher.home.tillo.ch using trusted certificate authorities within: /etc/kubernetes/ssl/certs/serverca"
time="2024-09-10T08:10:07Z" level=error msg="Could not securely connect to https://rancher.home.tillo.ch: Get \"https://rancher.home.tillo.ch\": tls: failed to verify certificate: x509: certificate signed by unknown authority"
```

As you can verify by yourself, https://rancher.home.tillo.ch has a valid certificate signed by Let's Encrypt.
The only reason it's failing is because the agent relies on /v3/settings/cacerts to confirm the CA, but behind my reverse proxy Rancher uses a private CA so this setting shows a CA that is different from Let's Encrypt.

I have the following options:

- Removing TLS offloading on my reverse proxy, exposing my private CA as well as Rancher's web service to the unforgiving bare Internet and lowering the public reputation of my public IP address
- Using Rancher without TLS internally, lowering the security of my internal traffic
- Setting up a VPN link between the Rancher agent environment and my internal network and a dedicated DNS record to reach Rancher through the tunnel, adding costs and operational complexity

I don't like any of these options, but I've noticed it's a common topic in public forums and no one has any alternative.
I believe that if the Rancher console works fine in a browser without any workaround, so should any Rancher agent.

I get why cacerts should be kept as-is in most environments, but by making it readOnly advanced configurations become impossible.
Maybe, instead of a rollback request, this should be a feature request where cacerts can be augmented by an optional cacerts-additional setting.

**Comment by tevaum** [comment]
@tillo, I'm running into this same issue. Do you mind sharing the detailed steps you used to "solve" this?
